### DISCUSSION TITLE
Error response code on Invalid or unauthorized call to Token endpoint

### DISCUSSION THREAD

**Discussion by sagarshah1983** [ory/hydra#2677] [Q&A] [answered]
Hello everyone,
We are using Hydra 1.9.
We are calling token endpoint in order to test compliance with OpenID Specs - https://openid.net/specs/openid-connect-core-1_0-final.html#ClientAuthentication

As per specs section 3.1.3.4:
```
If the Token Request is invalid or unauthorized, the Authorization Server constructs the error response. The parameters of the Token Error Response are defined as in Section 5.2 of OAuth 2.0 [RFC6749]. 
The HTTP response body uses the application/json media type with HTTP response code of 400.
```

Having said that, when we call token endpoint, we are seeing 401 Unauthorized response error code from the failed token endpoint.
curl --location --request POST 'https://{AUTHZ DNS}/oauth2/token' \
--header 'Authorization: Basic invalid-client-secret-base64-encoded' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'grant_type=client_credentials'

Can this be treated as bug?
Please advise and suggest your views.

Regards,
Sagar

**Comment by sagarshah1983** [comment, marked as answer]
I've created bug #2682 for the above reported issue

**Reply by vinckr** [reply]
Then I will mark this as answered for now and discussion continues in the issue.

**Reply by sagarshah1983** [reply]
Thanks @vinckr !
